src/Microsoft.OpenApi/Services/OpenApiWalker.cs

```diff
@@ -46,12 +48,15 @@ public void Walk(OpenApiDocument doc)
         /// <param name="tags"></param>
         internal void Walk(IList<OpenApiTag> tags)
         {
+            _visitor.Enter(OpenApiConstants.Tags);
```

Comment:
>    _visitor.Enter(OpenApiConstants.Tags); [](start = 8, length = 42)

Enter/Exit should be responsibility of the caller (e.g. in this case, it should go in the Walk(doc) ).

This will ensure that we don't do duplicate Enter calls for an object in an array (like the tags array).

Reply:
You already kinda handled this for Tag (but not for Server), but moving the Error/Exits to the caller altogether would make it more consistent. 



---
In reply to: [162477655](https://github.com/Microsoft/OpenAPI.NET/pull/176#discussion_r162477655) [](ancestors = 162477655)

Reply:
I opened another comment to discuss this.

---
In reply to: [162478481](https://github.com/Microsoft/OpenAPI.NET/pull/176#discussion_r162478481) [](ancestors = 162478481,162477655)